src/gluonts/mx/distribution/empirical_distribution.py

```diff
@@ -0,0 +1,258 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import Optional, Tuple
+
+import numpy as np
+import mxnet as mx
+
+from gluonts.core.component import validated
+from gluonts.mx import Tensor
+
+from .distribution import Distribution, getF
+from .distribution_output import DistributionOutput, ArgProj
+
+
+class EmpiricalDistribution(Distribution):
+    r"""
+    A class representing empirical distribution.
+    By default, the target is assumed to be vector-valued, i.e., `event_shape` is 1. However, note that each dimension
+    is assumed to be independent when computing variance and CRPS loss.
+
+    Also, for computing CDF and quantiles, it is assumede that samples are distinct along the samples dimension, which
+    should almost always be the case if samples are drawn from continuous distributions.
+
+    Parameters
+    ----------
+    samples
+        Tensor containing samples, of shape `(num_samples, *batch_shape, *event_shape)`.
+    event_dim
+        Number of event dimensions, i.e., length of the `event_shape` tuple.
+        This is `0` for distributions over scalars, `1` over vectors, `2` over matrices, and so on.
+    """
+
+    @validated()
+    def __init__(self, samples: Tensor, event_dim: int = 1) -> None:
```

Comment:
I’m not sure I understand this: why is 1 the default here? Should this not be a mandatory argument, without any default?

Reply:
ya, removed.